pkg/cmd/release-payload-controller/payload_verification_controller.go

```diff
@@ -0,0 +1,189 @@
+		return
+	}
+
+	go func() {
+		wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}()
+
+	<-ctx.Done()
+}
+
+func (c *PayloadVerificationController) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *PayloadVerificationController) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.sync(ctx, key.(string))
+
+	if err == nil {
+		c.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %w", key, err))
+	c.queue.AddRateLimited(key)
+
+	return true
+}
+
+func (c *PayloadVerificationController) sync(ctx context.Context, key string) error {
+	klog.V(4).Infof("Starting PayloadVerificationController sync")
+	defer klog.V(4).Infof("PayloadVerificationController sync done")
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the ReleasePayload resource with this namespace/name
+	originalReleasePayload, err := c.releasePayloadLister.ReleasePayloads(namespace).Get(name)
+	// The ReleasePayload resource may no longer exist, in which case we stop processing.
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// If there are any JobResults defined, then we don't need to do anything else here...
+	if len(originalReleasePayload.Status.BlockingJobResults) != 0 || len(originalReleasePayload.Status.InformingJobResults) != 0 {
+		klog.V(5).Infof("ReleasePayload: '%s/%s' already synced", originalReleasePayload.Namespace, originalReleasePayload.Name)
+		return nil
+	}
+
+	releasePayload := originalReleasePayload.DeepCopy()
+
+	klog.V(4).Infof("Syncing Payload Verification for ReleasePayload: %s/%s", releasePayload.Namespace, releasePayload.Name)
+	for _, verify := range releasePayload.Spec.PayloadVerificationConfig.BlockingJobs {
+		releasePayload.Status.BlockingJobResults = append(releasePayload.Status.BlockingJobResults, generateJobStatus(verify))
+	}
+
+	for _, verify := range releasePayload.Spec.PayloadVerificationConfig.InformingJobs {
+		releasePayload.Status.InformingJobResults = append(releasePayload.Status.InformingJobResults, generateJobStatus(verify))
+	}
+
+	releasePayload.Canonicalize()
+
+	if !reflect.DeepEqual(originalReleasePayload, releasePayload) {
+		_, err := c.releasePayloadClient.ReleasePayloads(releasePayload.Namespace).UpdateStatus(ctx, releasePayload, v1.UpdateOptions{})
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func generateJobStatus(config v1alpha1.CIConfiguration) v1alpha1.JobStatus {
+	aggregateState := v1alpha1.JobStatePending
+	if config.AnalysisJobCount > 0 {
+		aggregateState = v1alpha1.JobStateIgnored
```

Comment:
don't set this.  it belongs to the thing doing updates based on the prowjob results.

Reply:
Done